pkg/apis/resource/validation/validation_resourceslice_test.go

```diff
@@ -820,20 +900,104 @@ func createSharedCounters(count int) []resourceapi.CounterSet {
 	sharedCounters := make([]resourceapi.CounterSet, count)
 	for i := 0; i < count; i++ {
 		sharedCounters[i] = resourceapi.CounterSet{
-			Name:     fmt.Sprintf("sharedcounters-%d", i),
+			Name:     fmt.Sprintf("counterset-%d", i),
 			Counters: testCounters(),
 		}
 	}
 	return sharedCounters
 }
 
-func createConsumesCounter(count int) []resourceapi.DeviceCounterConsumption {
+func createConsumesCounters(count int) []resourceapi.DeviceCounterConsumption {
 	consumeCapacity := make([]resourceapi.DeviceCounterConsumption, count)
 	for i := 0; i < count; i++ {
 		consumeCapacity[i] = resourceapi.DeviceCounterConsumption{
-			SharedCounter: fmt.Sprintf("sharedcounters-%d", i),
-			Counters:      testCounters(),
+			CounterSet: fmt.Sprintf("counterset-%d", i),
+			Counters:   testCounters(),
 		}
 	}
 	return consumeCapacity
 }
+
+func TestResourceSliceSpecSize(t *testing.T) {
```

Comment:
Shall we perhaps remove this and focus instead on doing this kind of check via the integration test? I think it is a bit more complete over there.

Reply:
Agree with that. Just for now the integration tests are failing and I haven't had time to look into that yet. If you happen to know what the issue might be, please let me know.

Reply:
The taint key is missing in `testPublishResourceSlices`. One only sees the error when running with at least -v=3 - not ideal, we should add an "inform me about problems with my slices" Go API to the resourceslice controller in 1.34.

I'm a bit puzzled why that surfaces now. `validateDeviceTaints` was called, only the length check was missing.
Ah! I fixed the "all" integration test case such that it now enables the feature. That was missing before, so the faulty taint got dropped. :man_facepalming:

Reply:
I think I found the missing taint key, but not sure what needs to happen with the "all" integration test case.

I have updated the `NewMaxResourceSlice` function to reflect the changes and removed the max size unit test.

Reply:
I already fixed the "all" configuration in one of the changes that you copied from me into this PR - that is what caused test failure because with the fixed configuration, the apiserver no longer dropped the invalid field and complained instead.